src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorTranslateDiagnosticsService.cs

```diff
@@ -479,6 +480,18 @@ private bool TryGetOriginalDiagnosticRange(Diagnostic diagnostic, RazorCodeDocum
             MappingBehavior.Inferred,
             out originalRange))
         {
+            if (!IsCompilerDiagnostic(diagnostic))
+            {
+                if (diagnostic.Range.Start.TryGetAbsoluteIndex(sourceText, _logger, out var _) &&
+                    diagnostic.Range.End.TryGetAbsoluteIndex(sourceText, _logger, out var _))
+                {
+                    // We can't remap the range correctly, but it's probably an analyzer diagnostic so we still want to show it to the user
+                    // if it's still in the razor document range
```

Comment:
This is a very scary assumption. In a non-trivial razor file I would expect a lot of false positives here from errors in generated code that happens to be within the length of a razor document. Given we know of exactly one scenario that exists where there is actually a razor analyzer, and its not really a supported scenario, I would guess that most of the time this will be hit will actually be a false positive.

Can we tag the diagnostics in some way, with a custom tag that indicates if its from the .razor file and not the .razor.g.cs file? eg, based on the `identifier` parameter that is provided here: https://github.com/dotnet/roslyn/pull/71365/files#diff-7eaffa27e852935ff999303c53260d548c4a45c4f6e9901740df320dfcd229dfR33

_HOWEVER_ the reason I say that is because I don't know what a "Compiler" DiagnosticType implies. Does it, in fact, act as a way of tagging which diagnostics come from the Razor analyzer? What do IDE and CA diagnostics in the genereated file come back as?

Reply:
I need to check those, good call out. Will get to it in the pst am

The type is vs specific and compiler is hard code for cs###